pkg/controller/cloud/node_controller.go

```diff
@@ -284,6 +284,16 @@ func (cnc *CloudNodeController) AddCloudNode(obj interface{}) {
 			return err
 		}
 
+		providerID, err := cloudprovider.GetInstanceProviderID(cnc.cloud, types.NodeName(curNode.Name))
+		if err == nil {
+			curNode.Spec.ProviderID = providerID
+		} else {
+			// we should attempt to set providerID on curNode, but
+			// we can continue if we fail since we will attempt to set
+			// node addresses given the node name in getNodeAddressesByProviderIDOrName
+			glog.Errorf("%v", err)
```

Comment:
I also opted to not return here since some functions will try using the node name if the provider id fails. Example [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cloud/node_controller.go#L375-L385). To me it makes sense to fail loudly if you fail to get a provider id, but it shouldn't prevent nodes from removing the cloud taint and being left un-scheduable.